File: generator/src/java/darkness/generator/scripts/uka17/HorizontalLight.java
package darkness.generator.scripts.uka17;

import darkness.generator.api.BulbGroup;

public class HorizontalLight extends BaseScript {

	@Override
	public void run() {
		super.run();

		whiteover();
		whiteover();
		whiteover();
	}

	private void whiteover()
	{
		for (BulbGroup column : columns) {
			set(column, 255, 255, 255);
			skip(1);
		}
		for (BulbGroup column : columns) {
			set(column, 0, 0, 0);
			skip(2);
		}
	}
}

File: generator/src/kotlin/darkness/generator/scripts/uka17/HorizontalLight.kt
package darkness.generator.scripts.uka17

class HorizontalLight : BaseScript() {
    override fun run() {
        super.run()

        whiteover()
        whiteover()
        whiteover()
    }

    private fun whiteover() {
        for (column in columns) {
            set(column, 255, 255, 255)
            skip(1)
        }
        for (column in columns) {
            set(column, 0, 0, 0)
            skip(2)
        }
    }
}
